ask-sdk-core/src/com/amazon/ask/attributes/persistence/PersistenceAdapter.java

```diff
@@ -45,4 +45,14 @@
      */
     void saveAttributes(RequestEnvelope envelope, Map<String, Object> attributes) throws PersistenceException;
 
+    /**
+     * Deletes attributes from persistence. A {@link RequestEnvelope} is passed to the adapter so current request parameters
+     * can be used as an attribute key.
+     *
+     * @param envelope the current request envelope
+     * @param attributes attributes to persist
```

Comment:
No need for this. The RequestEnvelope should be sufficient to key the item to delete.